### PR TITLE
Use Enzyme (almost) everywhere

### PR DESCRIPTION
**Pull request by willdurand**
Refs #1104

---

I did not touch the following files:

- `tests/unit/core/testErrorHandler.js`: not easy to convert, next PR
- `tests/unit/ui/components/TestOverlayCard.js`: modified in #5730 
- `tests/unit/ui/components/TestSwitch.js`: to be removed
- `tests/unit/ui/components/TestRating.js`: might conflict with WIP
- `tests/unit/amo/components/TestRatingManager.js`:  might conflict with WIP
- `tests/unit/amo/components/TestLink.js`: next PR
- `tests/unit/core/components/TestInstallSwitch.js`: to be removed
- `tests/unit/core/components/TestAuthenticateButton.js`: next PR
- `tests/unit/core/components/TestInfoDialog.js`: not easy to convert, next PR-
- `tests/unit/disco/components/TestApp.js`: next PR

### CODE REVIEW

File: tests/unit/core/components/TestServerHtml.js [hunk 1]
```diff
@@ -39,135 +35,136 @@ describe(__filename, () => {
...
   it("doesn't render GA script when trackingEnabled is false", () => {
-    const html = findRenderedDOMComponentWithTag(
-      render({ trackingEnabled: false }),
-      'html',
-    );
-    const ga = html.querySelectorAll(
-      'script[src="https://www.google-analytics.com/analytics.js"]',
-    );
-    expect(ga.length).toEqual(0);
+    const root = render({ trackingEnabled: true });
```
Comment (rebmullin) [inline tests/unit/core/components/TestServerHtml.js]: should this be trackingEnabled: false?

File: tests/unit/core/components/TestServerHtml.js [hunk 2]
```diff
@@ -39,135 +35,136 @@ describe(__filename, () => {
...
+    const ga = root.find({
+      src: 'https://www.google-analytics.com/analytics.js',
+    });
+
+    expect(ga).toHaveLength(1);
```
Comment (rebmullin) [inline tests/unit/core/components/TestServerHtml.js]: should this be length 0 ?

File: tests/unit/core/components/TestServerHtml.js [hunk 3]
```diff
@@ -39,135 +35,136 @@ describe(__filename, () => {
...
     );
   });
 
   it('renders title', () => {
-    const html = findRenderedDOMComponentWithTag(render(), 'html');
-    const title = html.querySelector('title').textContent;
-    expect(title).toEqual('test title');
+    const root = render();
+
+    expect(root.find('title')).toHaveText('test title');
```
Comment (rebmullin) [inline tests/unit/core/components/TestServerHtml.js]: I know this is kind of how it was working before in the sense you could't override the title but is there a way to show that you could and that is the new title? Would that be useful here?

File: tests/unit/ui/components/TestShowMoreCard.js [hunk 4]
```diff
@@ -1,68 +1,84 @@
...
 
-    Simulate.click(rootNode.querySelector('.Card-footer-link a'));
+    // We cannot directly target the footer link, so we simulate the click here.
+    root.instance().onClick(createFakeEvent());
+    root.update();
 
-    expect(rootNode.className).toContain('ShowMoreCard--expanded');
-    expect(root.state.expanded).toEqual(true);
+    expect(root).toHaveClassName('ShowMoreCard--expanded');
+    expect(root).toHaveState('expanded', true);
```
Comment (rebmullin) [inline tests/unit/ui/components/TestShowMoreCard.js]: do we need both? Is showing the behavior change enough?

File: tests/unit/ui/components/TestShowMoreCard.js [hunk 5]
```diff
@@ -1,68 +1,84 @@
...
 
-    expect(rootNode.querySelector('.Card-footer-link')).toEqual(null);
+    expect(root).toHaveProp('footerLink', null);
   });
 
   it('is expanded by default', () => {
     const root = render({ children: 'Hello I am description' });
-    expect(root.state.expanded).toEqual(true);
+
+    expect(root).toHaveState('expanded', true);
```
Comment (rebmullin) [inline tests/unit/ui/components/TestShowMoreCard.js]: should we check the behavior (for existing class). I guess I was the impression that was better..let me know if I am missing something tho :/

File: tests/unit/core/components/TestServerHtml.js (hunk 1 above)
Comment (willdurand) [reply tests/unit/core/components/TestServerHtml.js]: oh good catch!

File: tests/unit/core/components/TestServerHtml.js (hunk 2 above)
Comment (willdurand) [reply tests/unit/core/components/TestServerHtml.js]: oh good catch!

File: tests/unit/core/components/TestServerHtml.js (hunk 3 above)
Comment (willdurand) [reply tests/unit/core/components/TestServerHtml.js]: I am not sure to know how to do it..

File: tests/unit/ui/components/TestShowMoreCard.js (hunk 4 above)
Comment (willdurand) [reply tests/unit/ui/components/TestShowMoreCard.js]: yep, makes sense to only keep the class name.

**Comment by willdurand** [comment]
thanks for the review @rebmullin!

**Pull request merged**